### PR TITLE
Implemented unique name on the backend for things.

### PR DESCRIPTION
**Pull request by IqbalAshif**

### CODE REVIEW

File: services/garage-backend/src/controllers/thingController.ts [hunk 1]
```diff
@@ -14,11 +14,16 @@ import { serializeThing } from '../serializers/things';
 // POST /things
 export const createThing = async (req: Request, res: Response, next: NextFunction) => {
   try {
+    const thingDocument = await Thing.findOne({ name: req.body.name });
+    if (thingDocument) {
+      next(new BadRequestError('Thing with same name exists in the Database'));
```
Comment (serushakov) [inline services/garage-backend/src/controllers/thingController.ts:19]:
```suggestion
      next(new BadRequestError('Thing with same name exists in the Database'));
      return;
```
don't forget to return

File: services/garage-backend/src/controllers/thingController.ts [hunk 2]
```diff
@@ -63,6 +68,10 @@ export const findThingById = async (req: Request, res: Response, next: NextFunct
 // PUT /things/:thingId
 export const updateThing = async (req: Request, res: Response, next: NextFunction) => {
   try {
+    const thingDocument = await Thing.findOne({ name: req.body.name });
+    if (thingDocument) {
+      next(new BadRequestError('Thing with same name exists in the Database'));
```
Comment (serushakov) [inline services/garage-backend/src/controllers/thingController.ts:73]:
```suggestion
      next(new BadRequestError('Thing with same name exists in the Database'));
      return;
```

File: services/garage-backend/src/models/Thing.ts [hunk 3]
```diff
@@ -12,6 +12,7 @@ export interface ThingDocument
 const thingSchema = new mongoose.Schema<ThingDocument>({
   name: {
     type: String,
+    index: { unique: true },
```
Comment (serushakov) [inline services/garage-backend/src/models/Thing.ts:15]: hmm, are you sure this is the way field is made unique? 
```suggestion
    unique: true,
```
I'm quite sure it's just `unique`

**Review by serushakov — CHANGES_REQUESTED** [review]
Looking good! Good that you also remembered to modify `update` method

**Review by serushakov — APPROVED** [review]

**Review by mjcarterdev — APPROVED** [review]

**Pull request merged**